src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -29,23 +30,23 @@ public enum TagName {
      *
      * @param tagName A valid tag name.
      */
-    public Tag(String tagName) {
+    public Tag(TagName tagName) {
         requireNonNull(tagName);
-        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
-        this.tagName = TagName.valueOf(tagName.toUpperCase());
+        this.tagName = tagName;
     }
 
     /**
      * Returns true if a given string is a valid tag name.
      */
     public static boolean isValidTagName(String test) {
```

Comment:
We should not use Try-Catch block as conditional branch and return values, if I rmb correctly